Don't allow player to be registered twice.

jodator opened the pull request:
Don't allow player to be registered twice.

jodator committed:
Don't allow player to be registered twice.

pjasiun commented on app/command/registration.js:
```diff
@@ -46,19 +46,23 @@ class Registration {
+			// Add the player only if he/she is not registered already.
+			// We could return if user in in registered pool already but it could be abused to check if someone else is already registered.
+			if ( registered.indexOf( user ) === -1 ) {
+				registered.push( user );
```
`registered` is last 20 registered users. The plan was to handle removing player from the previous match, not only the last one. It means that with this code user can not registered if he was registered in any of the last 5 matches.

jodator replied:
Actually your plan is invalid because of:

https://github.com/pjasiun/fuma/blob/master/app/command/registration.js#L57

Registered will never be > 4 :dancer: (I've wrote test for that).

I'm updating this PR with tests...

jodator commented:
OK, I'll fix that later on. I was doing this with my morning coffee so I've missed that :D

jodator committed:
Added tests for Registration command.

jodator commented:
OK, this must be merged. I've added tests & this command have 100% coverage.
For this to work I've revamped main handler a bit and moved `asyncResponse` there (this will be preferable way of passing asyn answer as it is easier to write tests (Inversion of Control).
After that is merged I can work on tests for other commands :)
